src/cloud_run/tables/table_manager.py

```diff
@@ -0,0 +1,42 @@
+from google.api_core.exceptions import GoogleAPIError
+
+from utils.constants import PROJECT_ID, DATASET_ID, KEYS
+from utils.table_logger import setup_logger
+
+
+class TableManager:
+
+    def __init__(self, client, table, row):
+        self.bq_client = client
+        self.table = table
+        self.row = row
+        self.key_field = KEYS[self.table]
+        self.key_value = self.row[KEYS[self.table]]
+
+    def not_exist(self, key):
+        table = f"{PROJECT_ID}.{DATASET_ID}.{self.table}"
+        query = f"""
+            SELECT COUNT(*) AS record_count
+            FROM `{table}`
+            WHERE {self.key_field} = {key}
+        """
+        query_result = self.bq_client.query(query).result()
+        record_count = next(query_result).get("record_count", 0)
+
+        if record_count > 0:
```

Comment:
can do one liner here: `return record_count == 0` (assuming negative value would not happen)

Reply:
brilliant